zookeeper-server/src/main/java/org/apache/zookeeper/ZooKeeper.java

```diff
@@ -3028,6 +3147,104 @@ public void removeAllWatches(String path, WatcherType watcherType, boolean local
         removeWatches(ZooDefs.OpCode.removeWatches, path, null, watcherType, local, cb, ctx);
     }
 
+    /**
+     * Modes available to {@link ZooKeeper#addWatch(String, Watcher, AddWatchMode)}
+     */
+    public enum AddWatchMode {
```

Comment:
What about moving this enum  to the top level (not an inner class) ?

Reply:
Makes no different to me. You prefer top level?

Reply:
yes, this way if will be less tied to the ZooKeeper class, not a big deal.

Reply:
done